File: android/kotlin-davinci/davinci/src/main/java/com/pingidentity/samples/app/davinci/collector/SubmitButton.kt
/*
 * Copyright (c) 2024 - 2025 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.samples.app.davinci.collector

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.wrapContentWidth
import androidx.compose.material3.Button
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.pingidentity.davinci.collector.SubmitCollector

/**
 * The submit button.
 *
 * @param field The submit collector.
 * @param onNext The callback to be called when the next node is triggered.
 */
@Composable
fun SubmitButton(
    field: SubmitCollector,
    onNext: () -> Unit,
) {
    field.value = ""
    Row(
        modifier =
        Modifier
            .padding(4.dp)
            .fillMaxWidth(),
    ) {
        Button(
            modifier =
            Modifier
                .fillMaxWidth()
                .wrapContentWidth(Alignment.CenterHorizontally),
            onClick = {
                field.value = "submit"
                onNext()
            },
        ) {
            androidx.compose.material3.Text(field.label)
        }
    }
}

File: javascript/reactjs-todo-davinci/client/components/davinci-client/submit-button.js
/*
 * ping-sample-web-react-davinci
 *
 * submit-button.js
 *
 * Copyright (c) 2025 - 2026 Ping Identity Corporation. All rights reserved.
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */
import React from 'react';

export default function SubmitButton({ collector, isLoading }) {
  return (
    <button type="submit" className="btn btn-primary w-100 mb-2" disabled={isLoading}>
      {
        /**
         * Render a small spinner during submission calls
         */
        isLoading ? (
          <span
            className="spinner-border spinner-border-sm"
            role="status"
            aria-hidden="true"
          ></span>
        ) : null
      }
      <span>{collector.output.label}</span>
    </button>
  );
}
